### PR TITLE
[Order] Visibility rules refactoring

### PR DESCRIPTION
**Pull request by dobryy**

### CODE REVIEW

File: rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php [hunk 1]
```diff
@@ -100,63 +98,28 @@ public function getNodeTypes(): array
...
-            $classMethods[$classMethodName]['abstract'] = $classStmt->isAbstract();
-            $classMethods[$classMethodName]['final'] = $classStmt->isFinal();
-            $classMethods[$classMethodName]['static'] = $classStmt->isStatic();
-            $classMethods[$classMethodName][self::POSITION] = $position;
-        }
-
-        $sortedMethods = $this->getMethodsSortedByVisibility($classMethods);
-        $methodsInPreferredOrder = $this->getMethodsInPreferredOrder($sortedMethods);
+        $currentMethodsOrder = $this->stmtOrder->getStmtsOfTypeOrder($node, ClassMethod::class);
+        $methodsInDesiredOrder = $this->getMethodsInDesiredOrder($node);
```
Comment (TomasVotruba) [inline rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php:106]: :+1:

File: rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php [hunk 2]
```diff
@@ -100,63 +98,28 @@ public function getNodeTypes(): array
...
-
-        return array_keys($classMethods);
+        $classMethods = $this->stmtVisibilitySorter->sortMethods($classLike);
+        $classMethods = array_keys($classMethods);
+
+        return $this->applyPreferredPosition($classMethods);
     }
 
-    private function getMethodsInPreferredOrder(array $sortedMethods): array
+    private function applyPreferredPosition(array $classMethods): array
```
Comment (TomasVotruba) [inline rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php:121]: `array` would deserve types

I'm going to re-enable this in PHPStan, so it's covered by CI

**Comment by TomasVotruba** [comment]
Just the `array` types. The rest looks good to me

File: rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php (hunk 2 above)
Comment (TomasVotruba) [reply rules/order/src/Rector/ClassLike/OrderMethodsByVisibilityRector.php:121]: https://github.com/rectorphp/rector/pull/3936

**Comment by dobryy** [comment]
> Just the `array` types. The rest looks good to me

Ready.
array<int,string> - when index is not continuous value from 0 to N.
string[] - when the array is an indexed array with the continuous index value. 

Let me know if it makes sense :)

**Comment by TomasVotruba** [comment]
All good, thanks

**Pull request merged**